src/plone/recipe/zope2instance/__init__.py

```diff
@@ -244,6 +244,23 @@ def build_zope_conf(self):
                 environment_vars = '\n'.join(["%s %s" % (env_var[0], env_var[1])
                                              for env_var in env_vars])
             environment_vars = environment_template % environment_vars
+        cgi_environment_vars = options.get('cgi-environment-vars', '')
+        if cgi_environment_vars:
+            # if the vars are all given on one line we need to do some work
+            if not '\n' in cgi_environment_vars:
+                keys = []
+                values = []
+                env_vars = cgi_environment_vars.split()
+                # split out the odd and even items into keys, values
+                for var in env_vars:
+                    if divmod(env_vars.index(var) + 1, 2)[1]:
+                        keys.append(var)
+                    else:
+                        values.append(var)
+                env_vars = zip(keys, values)
+                cgi_environment_vars = '\n'.join(["%s %s" % (env_var[0], env_var[1])
+                                             for env_var in env_vars])
```

Comment:
It looks like this processing could easily be factored into a function to avoid duplicating code between environment_vars and cgi_environment_vars.